src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -417,7 +417,9 @@ extern "C" void QCALLTYPE RuntimeMethodHandle_InvokeMethod(
     GCStress<cfg_any>::MaybeTrigger();
 
     FrameWithCookie<ProtectValueClassFrame> *pProtectValueClassFrame = NULL;
+    FrameWithCookie<ProtectValueClassFrame>* pProtectValueClassFrame2 = NULL;
```

Comment:
You should not need second list and ProtectValueClassFrame. It should be enough to just append a new entry to the existing `pValueClasses` list.